src/evergreen/api.py

```diff
@@ -198,6 +216,11 @@ def _raise_for_status(response: requests.Response) -> None:
         try:
             json_data = response.json()
             if response.status_code >= 400 and "error" in json_data:
+                LOGGER.error(
+                    "Response received",
+                    response_status_code=response.status_code,
+                    response_text=response.text,
+                )
```

Comment:
This would be the statement that would be behind the flag. Because this is an error level statement, it's likely to actually be seen and included by our default configurations. So, to prevent noisiness, this flag would guard against it.

This would mean that we would have to supplement this log statement with the request information as well if we just wanted to always see the request and response for all error only calls.